filters/auth/tokeninfocache.go

```diff
@@ -58,35 +65,21 @@ func (c *tokeninfoCache) getTokeninfo(token string, ctx filters.FilterContext) (
 }
 
 func (c *tokeninfoCache) cached(token string) map[string]any {
-	now := c.now()
-
-	c.mu.Lock()
-
-	if e, ok := c.cache[token]; ok {
+	if v, ok := c.cache.Load(token); ok {
+		now := c.now()
+		e := v.(*entry)
 		if now.Before(e.expiresAt) {
-			c.history.MoveToFront(e.href)
-			cachedInfo := e.info
-			c.mu.Unlock()
-
 			// It might be ok to return cached value
 			// without adjusting "expires_in" to avoid copy
 			// if caller never modifies the result and
 			// when "expires_in" did not change (same second)
 			// or for small TTL values
-			info := shallowCopyOf(cachedInfo)
+			info := maps.Clone(e.info)
```

Comment:
Is this ok for 2 goroutines access the same e.info ?
I tried to check the docs but it doesn't tell about concurrent access is fine.
A different way could be to use also sync.Map for info member instead of map.

Reply:
> Is this ok for 2 goroutines access the same e.info ?

Is it a problem if both are readonly?

Reply:
2 reads would be fine but later it will surely write/replace the object.

Reply:
Ah no it's read only 😀